cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/handler/async/runnable/RightsizeRunnable.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2021 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.servlet.handler.async.runnable;
+
+import com.linkedin.kafka.cruisecontrol.KafkaCruiseControl;
+import com.linkedin.kafka.cruisecontrol.analyzer.ProvisionRecommendation;
+import com.linkedin.kafka.cruisecontrol.analyzer.ProvisionStatus;
+import com.linkedin.kafka.cruisecontrol.common.Utils;
+import com.linkedin.kafka.cruisecontrol.config.KafkaCruiseControlConfig;
+import com.linkedin.kafka.cruisecontrol.config.constants.AnomalyDetectorConfig;
+import com.linkedin.kafka.cruisecontrol.detector.Provisioner;
+import com.linkedin.kafka.cruisecontrol.detector.ProvisionerState;
+import com.linkedin.kafka.cruisecontrol.detector.RightsizeOptions;
+import com.linkedin.kafka.cruisecontrol.servlet.parameters.RightsizeParameters;
+import com.linkedin.kafka.cruisecontrol.servlet.response.RightsizeResult;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Supplier;
+
+import static com.linkedin.kafka.cruisecontrol.detector.AnomalyDetectorUtils.KAFKA_CRUISE_CONTROL_OBJECT_CONFIG;
+
+
+/**
+ * The async runnable for running a provision rightsize request
+ */
+public class RightsizeRunnable extends OperationRunnable {
+  private static final String RECOMMENDER_UP = "Recommender-Under-Provisioned";
+  protected final RightsizeParameters _parameters;
+
+  public RightsizeRunnable(KafkaCruiseControl kafkaCruiseControl, OperationFuture future, RightsizeParameters parameters) {
+    super(kafkaCruiseControl, future);
+    _parameters = parameters;
+  }
+
+  @Override
+  protected RightsizeResult getResult() throws Exception {
+    KafkaCruiseControlConfig config = _kafkaCruiseControl.config();
+    Map<String, Object> overrideConfigs;
+    overrideConfigs = Collections.singletonMap(KAFKA_CRUISE_CONTROL_OBJECT_CONFIG, _kafkaCruiseControl);
+    Provisioner provisioner = config.getConfiguredInstance(AnomalyDetectorConfig.PROVISIONER_CLASS_CONFIG,
+                                                           Provisioner.class,
+                                                           overrideConfigs);
+    Supplier<Set<String>> topicNameSupplier = () -> _kafkaCruiseControl.kafkaCluster().topics();
+    Set<String> topicNamesMatchedWithPattern = Utils.getTopicNamesMatchedWithPattern(_parameters.topic(), topicNameSupplier);
+    String topicName;
+    if (topicNamesMatchedWithPattern.size() > 1) {
+      throw new IllegalArgumentException("The number of topics provided to provision exceeds 1.");
```

Comment:
Should we rephrase this to
1. explain that this endpoint does not support rightsizing for multiple topics, and
2. indicate those multiple topics the request tried to rightsize?